src/Facility.CodeGen.JavaScript/JavaScriptGenerator.cs

```diff
@@ -137,248 +142,251 @@ public override CodeGenOutput GenerateOutput(ServiceInfo service)
-											}
-										}
-										else
-										{
-											if (validResponse.NormalFields!.Count == 0)
-											{
-												code.WriteLine("value = {};");
 											}
 											else
 											{
-												using (code.Block("if (result.json) {", "}"))
+												if (validResponse.NormalFields!.Count == 0)
+												{
+													code.WriteLine("value = {};");
+												}
+												else
 												{
-													code.WriteLine("value = result.json" + IfTypeScript($" as {responseValueType} | null") + ";");
+													using (code.Block("if (result.json) {", "}"))
+													{
+														code.WriteLine("value = result.json" + IfTypeScript($" as {responseValueType} | null") + ";");
+													}
 												}
 											}
 										}
+										code.WriteLine("}");
 									}
-									code.WriteLine("}");
-								}
 
-								using (code.Block("if (!value) {", "}"))
-									code.WriteLine("return createResponseError(status, result.json)" + IfTypeScript($" as IServiceResult<I{capMethodName}Response>") + ";");
+									using (code.Block("if (!value) {", "}"))
+										code.WriteLine("return createResponseError(status, result.json)" + IfTypeScript($" as IServiceResult<I{capMethodName}Response>") + ";");
 
-								if (httpMethodInfo.ResponseHeaderFields.Count != 0)
-								{
-									code.WriteLine("let headerValue" + IfTypeScript(": string | null | undefined") + ";");
-									foreach (var httpHeaderField in httpMethodInfo.ResponseHeaderFields)
+									if (httpMethodInfo.ResponseHeaderFields.Count != 0)
 									{
-										code.WriteLine($"headerValue = result.response.headers.get('{httpHeaderField.Name}');");
-										using (code.Block("if (headerValue != null) {", "}"))
-											code.WriteLine($"value.{httpHeaderField.ServiceField.Name} = {ParseFieldValue(httpHeaderField.ServiceField, service, "headerValue")};");
+										code.WriteLine("let headerValue" + IfTypeScript(": string | null | undefined") + ";");
+										foreach (var httpHeaderField in httpMethodInfo.ResponseHeaderFields)
+										{
+											code.WriteLine($"headerValue = result.response.headers.get('{httpHeaderField.Name}');");
+											using (code.Block("if (headerValue != null) {", "}"))
+												code.WriteLine($"value.{httpHeaderField.ServiceField.Name} = {ParseFieldValue(httpHeaderField.ServiceField, service, "headerValue")};");
+										}
 									}
-								}
 
-								code.WriteLine("return { value: value };");
+									code.WriteLine("return { value: value };");
+								}
 							}
 						}
-					}
 
-					if (TypeScript)
-					{
-						code.WriteLine();
-						code.WriteLine("private _fetch: IFetch;");
-						code.WriteLine("private _baseUri: string;");
+						if (TypeScript)
+						{
+							code.WriteLine();
+							code.WriteLine("private _fetch: IFetch;");
+							code.WriteLine("private _baseUri: string;");
+						}
 					}
-				}
-			}));
+				}));
+			}
 
 			if (Express)
```

Comment:
This code might be obsolete, but do you know if it will work with `NoHttp`?

Reply:
I'm not sure, but I can add a test to confirm and add handling here if necessary. If it's not compatible, the simplest thing would be to just make sure that `Express` and `NoHttp` are not used together yeah?